packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,46 @@
+import { type TSESLint, type TSESTree } from '@typescript-eslint/utils';
+
+import { createRule, getParserServices } from '../util';
+
+type MessageIds = 'forbiddenFunctionSpread';
+
+export default createRule<[], MessageIds>({
+  defaultOptions: [],
+  name: 'no-misused-spread',
+  meta: {
+    docs: {
+      description:
+        "Disallow spread operator that shouldn't be spread most of the time",
+      recommended: 'stylistic',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      forbiddenFunctionSpread:
+        'Spreading a function is almost always a mistake. Did you forget to call the function?',
+    },
+    schema: [],
+    type: 'suggestion',
+  },
+  create: (context): TSESLint.RuleListener => {
+    const listener = (node: TSESTree.SpreadElement): void => {
+      const services = getParserServices(context);
+      const checker = services.program.getTypeChecker();
+
+      const tsNode = services.esTreeNodeToTSNodeMap.get(node.argument);
+      const type = checker.getTypeAtLocation(tsNode);
```

Comment:
[Cleanup] We have a new thing for this 😄 https://typescript-eslint.io/blog/announcing-typescript-eslint-v6#type-checker-wrapper-apis

```suggestion
      const type = services.getTypeAtLocation(node.argument);
```